casper/src/main/scala/coop/rchain/casper/BlockCreator.scala

```diff
@@ -42,6 +42,19 @@ object BlockCreator {
     } yield isActive
   }
 
+  private def isBonded[F[_]: Sync](
+      block: BlockMessage,
+      runtimeManager: RuntimeManager[F],
+      publicKey: PublicKey
+  ): F[Boolean] = {
+    val lastProposedTuplespace = ProtoUtil.postStateHash(block)
+    for {
+      bonds     <- runtimeManager.computeBonds(lastProposedTuplespace)
+      validator = ByteString.copyFrom(publicKey.bytes)
+      isBonded  = bonds.exists(b => b.stake > 0 && b.validator == validator)
+    } yield isBonded
+  }
```

Comment:
If we return a `Map[Validator, Stake]` we need to call it only once even if the block has multiple parents. This is because each call to `computeBonds` is about 300ms.

Reply:
I take this back, we need bonds for different roots so we need to call this for each. Sorry, my mistake.